.changeset/great-scissors-boil.md

```diff
@@ -0,0 +1,5 @@
+---
+'@urql/exchange-graphcache': patch
+---
+
+list-items are never nullable when we aren't using schema-awareness, this would otherwise lead to scenario's where we don't refetch partial results. Formerly we made the assumption that a list-item was nullable when we weren't provided with a schema, this meant that if we'd provide `[undefined, 'entity:x']` from a resolver we wouldn't fetch the result but we'd return it.
```

Comment:
I'd avoid phrasing in changelogs that are long and don't state what the actual change is, i.e. "x is never y" is a statement and when skimming a reader can't be sure whether that was true before or is now true after the change. Instead, we can tell them directly what the issue was. So I'd prefer prefixing this with "Fix" to clarify that we had an unintentional behaviour in Graphcache.

```suggestion
Fix list items being returned as `null` even for non-nullable lists, when the entities are missing in the cache. This could happen when a resolver was added returning entities or their keys. This behaviour is now (correctly) only applied to partial results with schema awareness.
```